src/app/src/api/boundaries.js

```diff
@@ -48,6 +48,24 @@ const boundaryApi = api.injectEndpoints({
             }),
             invalidatesTags: getUpdateItemTagInvalidator(TAGS.BOUNDARY),
         }),
+
+        uploadReferenceImage: build.mutation({
+            query: newSubmission => ({
+                url: `/boundaries/${newSubmission.boundary}/reference-images/`,
+                method: 'POST',
+                data: newSubmission,
+            }),
+            invalidatesTags: getUpdateItemTagInvalidator(TAGS.BOUNDARY),
+        }),
+
+        updateReferenceImage: build.mutation({
+            query: updatedSubmission => ({
+                url: `/boundaries/${updatedSubmission.boundary}/reference-images/${updatedSubmission.id}/`,
+                method: 'PUT',
+                data: updatedSubmission,
+            }),
+            invalidatesTags: getUpdateItemTagInvalidator(TAGS.BOUNDARY),
+        }),
```

Comment:
A couple of things here:
- I don't think `newSubmission` makes as much sense here. Perhaps we can destructure some of these values.
- We shouldn't be using `BOUNDARY` tags for reference images unless we want them to be invalidated together. Since `query` has different arguments than boundaries, we also can't use the helper function for the update endpoint. Maybe at this point, we omit the tags and let the components re-fetch after they've been unmounted.
- There should be endpoints for querying the reference images. 
- I think there's enough going on here that reference images could have their own file in `/src/app/src/api`. 

```suggestion
        getReferenceImages: build.query({
            query: boundaryId => ({
                url: `/boundaries/${boundaryId}/reference-images/`,
                method: 'GET',
            }),
        }),

        uploadReferenceImage: build.mutation({
            query: ({ boundaryId, ...details }) => ({
                url: `/boundaries/${boundaryId}/reference-images/`,
                method: 'POST',
                data: details,
            }),
        }),

        updateReferenceImage: build.mutation({
            query: ({ boundaryId, referenceImageId, ...details }) => ({
                url: `/boundaries/${boundaryId}/reference-images/${referenceImageId}/`,
                method: 'PUT',
                data: details,
            }),
        }),
```

Reply:
Yeah, happy to have a look at this tomorrow. I was assuming we had to invalidate the boundary because we weren't going to have an endpoint for querying the images. So several things could change here.